packages/sdk/contractkit/src/celo-tokens.ts

```diff
@@ -101,16 +127,46 @@ export class CeloTokens {
   async forEachCeloToken<T>(
     fn: (info: CeloTokenInfo) => T | Promise<T>
   ): Promise<EachCeloToken<T>> {
-    const wrapperInfos = await Promise.all(
-      Object.values(celoTokenInfos).map(async (info: CeloTokenInfo) => {
-        const fnResult = fn(info)
-        return {
-          symbol: info.symbol,
-          data: fnResult instanceof Promise ? await fnResult : fnResult,
-        }
-      })
-    )
-    return wrapperInfos.reduce(
+    const wrapperInfoFunction = async () =>
+      Promise.all(
+        (await this.validCeloTokenInfos()).map(async (info) => {
+          const fnResult = fn(info)
+          return {
+            symbol: info.symbol,
+            data: fnResult instanceof Promise ? await fnResult : fnResult,
```

Comment:
can you just do `await Promise.resolve(fnResult)` or even `await fnResult`? I think awaiting a non promise is totally fine.

Reply:
Oh, yes, this was from the old code, totally missed it. Thanks